### PR TITLE
[DO NOT MERGE - WIP] Add Private Provider Docs

### PR DESCRIPTION
**Pull request by laurapacilio**
Users can now add private providers to the Terraform Cloud private registry. At this time, private providers are not available in Terraform Cloud. Once a provider, provider version, and one or more provider platforms have been added through the API, users can view the provider in the Terraform Cloud UI and begin using it in configurations. 

To support this new feature launch, this PR makes the following documentation updates:
- Adds a new Publishing Private Providers page that will help users understand the process and requirements for adding a private provider. It will point users to the API endpoints (documented in the API page) that they need to manage providers. We don't duplicate the documentation on the API page.
- Updates the other private and public registry documentation accordingly. 
- Renames the publishing modules page (from publish.mdx --> publish-modules.mdx) for clarity and adds a redirect. 

------------
**TODOs:**
- Add links to new private provider endpoints once docs are finalized.
- Check the providers page again once UI changes are done.

------

I have:

- [ ] Built the site locally or verified that the website preview successfully deployed. 
- [ ] Verified that images and formatting render correctly for each changed page.
- [x] Added redirects for moved or deleted pages.
- [x] Updated the relevant sidebar navigation files for added, deleted, reordered, or renamed pages.
- [x] Reviewed this content for adherence to [Engineering](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md) and [Learn](https://github.com/hashicorp/learn/blob/master/STYLE_GUIDE.md) style guides, especially for:
    - [x] Images are 2048 px wide, annotation color and format, images have alt text for accessibility
    - [x] UI elements are bolded
    - [ ] Code is properly formatted
- [ ] Performed a final copyedit to catch typos, punctuation, and other minor inconsistencies.
- [x] Discussed this content with engineering or tagged an engineering reviewer for technical accuracy.
- [x] Added one or more labels that describe both the type of change (e.g. clarification) and associated product (e.g. tfc). 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201758754579181

### CODE REVIEW

File: content/cloud-docs/registry/publish-providers.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+GPG signing is required for custom providers, and you must upload the public key of the GPG keypair used to sign the release. Refer to [Preparing and Adding a Signing Key](/registry/providers/publishing#preparing-and-adding-a-signing-key) for more details. Unlike the public Terraform Registry, the private registry does not automatically upload new releases. You must manually add new provider versions.
+
+## Publishing a New Provider
+
+You can publish and manage private providers through the Terraform Cloud API. Before you begin, confirm that you can access the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release. You will upload these artifacts in the following steps.
+
+To publish a new provider:
+
+1. Use the [Create a GPG key endpoint](link to endpoint) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx:39]: Just to correlate with my pending changes

```suggestion
1. Use the [Add a GPG key endpoint](link to endpoint) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```

**Review by madaleines — APPROVED** [review]
LGTM, going to pass to the team to also take a look

File: content/registry/providers/publishing.mdx [hunk 2]
```diff
@@ -102,13 +102,11 @@ GoReleaser is a tool for building Go projects for multiple platforms, creating a
 
 #### Manually Preparing a Release
```
Comment (laurapacilio) [inline content/registry/providers/publishing.mdx:103]: @rberecka and @madaleines I made a few adjustments here - could you review?

File: content/registry/providers/publishing.mdx [hunk 3]
```diff
@@ -128,14 +126,19 @@ Before publishing a provider, you must first sign in to the Terraform Registry w
...
-- Generate a GPG key to be used when signing releases (See [GitHub's detailed instructions](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-gpg-key) for help with this step, but you do not need to add the key to GitHub)
-- Export your public key in ASCII-armor format using the following command, substituting the GPG key ID created in the step above:
+To prepare and add a signing key:
 
-```console
-$ gpg --armor --export "{Key ID or email address}"
-```
+1. Generate a GPG key to be used when signing releases. Refer to [GitHub's detailed instructions](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-gpg-key) for this step, but you do not need to add the key to GitHub.
+
+  -> **Note:** The Terraform Registry API accepts both RSA and DSA keys, but not the default ECC type.
```
Comment (rberecka) [inline content/registry/providers/publishing.mdx:133]: 👍

**Review by duplo83 — COMMENTED** [review]
It all looks good to me!

File: content/cloud-docs/registry/publish-providers.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+GPG signing is required for custom providers, and you must upload the public key of the GPG keypair used to sign the release. Refer to [Preparing and Adding a Signing Key](/registry/providers/publishing#preparing-and-adding-a-signing-key) for more details. Unlike the public Terraform Registry, the private registry does not automatically upload new releases. You must manually add new provider versions and the associated release files.
+
+## Publishing a Provider and Creating a Version
+
+You must create a provider, at least one version, and at least one platform for that version before consumers can use your provider in configurations. Before you begin, confirm that you can access the provider binaries, public GPG signing key, `SHA256SUMS` file, and `SHA256SUMS.sig` file from at least one release. You will upload these artifacts in the following steps.
+
+To publish a new provider through the Terraform Cloud API:
+
+1. Use the [Upload a GPG key endpoint](/cloud-docs/api-docs/providers#upload-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx:39]:
```suggestion
1. Use the [Add a GPG key endpoint](/cloud-docs/api-docs/providers#add-a-gpg-key-private-provider) to add the public key that matches the signing key for the release. The response contains a `key-id` that you will use to create a provider version.
```

File: content/cloud-docs/registry/publish-providers.mdx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    ```
+
+1. Upload the `SHA256SUMS` and `SHA256SUMS.sig` files to the URLs returned in the previous step. The example command below uploads the files from a local machine.
+
+    ```
+    $ curl -T local-example/terraform-provider-random_3.1.3_SHA256SUMS
+      https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63
+    ```
+
+1. Use the [Upload a Provider Platform](/cloud-docs/api-docs/providers#upload-a-provider-platform-private-provider) endpoint to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```
Comment (madaleines) [inline content/cloud-docs/registry/publish-providers.mdx:60]:
```suggestion
1. Use the [Add a Provider Platform](/cloud-docs/api-docs/providers#add-a-provider-platform-private-provider) endpoint to create a platform for the version. Platforms are binaries that allow the provider to run on a particular operating system and architecture combination (e.g., Linux and AMD64). The response includes a `provider-binary-upload` URL that you will use to upload the binary file for the platform.
```

File: content/cloud-docs/api-docs/modules.mdx [hunk 6]
```diff
@@ -582,7 +590,7 @@ curl \
 }
 ```
 
-## Upload a Module Version (private module)
+## Upload a Module Version (Private Module)
```
Comment (madaleines) [inline content/cloud-docs/api-docs/modules.mdx:593]:
```suggestion
## Add a Module Version (Private Module)
```

**Review by madaleines — CHANGES_REQUESTED** [review]
I think we should change "Upload" to "Add" for consistency and to make links work properly

**Review by joekarl — APPROVED** [review]

**Review by laurenrother — APPROVED** [review]
Took a read through. These all look great 👍

File: content/cloud-docs/api-docs/private-registry/providers.mdx [hunk 7]
```diff
@@ -282,8 +356,55 @@ curl \
...
+    "id": "prov-cmEmLstBfjNNA9F3",
+    "type": "registry-providers",
+    "attributes": {
+      "name": "aws",
+      "namespace": "hashicorp",
+      "version-statuses": [
+        {
+          "version": "3.1.1"
+        }
+      ],
```
Comment (jbonhag) [inline content/cloud-docs/api-docs/private-registry/providers.mdx:373]: Could we please remove the `version-statuses` attribute from this response?

File: content/cloud-docs/api-docs/private-registry/providers.mdx [hunk 8]
```diff
@@ -222,7 +247,46 @@ curl \
...
+```json
+{
+  "data": {
+    "id": "prov-cmEmLstBfjNNA9F3",
+    "type": "registry-providers",
+    "attributes": {
+      "name": "aws",
+      "namespace": "hashicorp",
+      "registry-name": "private",
+      "version-statuses": [],
```
Comment (jbonhag) [inline content/cloud-docs/api-docs/private-registry/providers.mdx:261]: Could we please remove the `version-statuses` attribute from this response?

File: content/cloud-docs/registry/publish-providers.mdx (hunk 5 above)
Comment (laurapacilio) [reply content/cloud-docs/registry/publish-providers.mdx]: i got you! I'll make sure this is updated :)

Comment (madaleines) [reply content/cloud-docs/registry/publish-providers.mdx]: 🙌

File: content/cloud-docs/api-docs/private-registry/providers.mdx (hunk 8 above)
Comment (laurapacilio) [reply content/cloud-docs/api-docs/private-registry/providers.mdx]: done!

File: content/cloud-docs/api-docs/private-registry/providers.mdx (hunk 7 above)
Comment (laurapacilio) [reply content/cloud-docs/api-docs/private-registry/providers.mdx]: done!

File: content/cloud-docs/agents/changelog.mdx [hunk 9]
```diff
@@ -16,6 +16,21 @@ within each release are categorized into one or more of the following labels:
 Each version below corresponds to a release artifact available for download on
 the official [releases website](https://releases.hashicorp.com/tfc-agent/).
 
+## 1.1.0 (02/23/2022)
```
Comment (madaleines) [inline content/cloud-docs/agents/changelog.mdx:19]: should we update the date?

File: content/cloud-docs/api-docs/modules.mdx (hunk 6 above)
Comment (laurapacilio) [reply content/cloud-docs/api-docs/modules.mdx:593]: updated!

**Comment by laurapacilio** [comment]
closing because the branch got a bit wonky :-) New PR here - https://github.com/hashicorp/terraform-website/pull/2162

**Pull request closed without merging**